src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -9,34 +9,48 @@ add_dependencies(tests mgr-dashboard-test-venv)
 
 if(WITH_MGR_DASHBOARD_FRONTEND AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64|arm|ARM")
 
-set(mgr-dashboard-nodeenv ${CMAKE_CURRENT_BINARY_DIR}/node-env)
+if(NOT FREEBSD)
```

Comment:
i'd suggest add an option named `USE_SYSTEM_NPM` instead making FREEBSD an exception.

Reply:
@tchaikov 
Like this?
How also joined the 2 blocks of building command per version of HAVE_SYSTEM_NPM as cleanup